code/modules/reagents/newchem/medicine.dm

```diff
@@ -0,0 +1,711 @@
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		H.hulk_time = 0
+	var/needs_update = M.mutations.len > 0
+	for(var/datum/dna/gene/G in dna_genes)
+		if(G.is_active(M))
+			if(G.name == "Hulk" && ishuman(M))
+				G.OnMobLife(M)
+			var/tempflag = 0
+			if(ishuman(M))
+				var/mob/living/carbon/human/H = M
+				if(H.species && (G.block in H.species.default_blocks))
+					tempflag |= GENE_NATURAL
+			if(G.can_deactivate(M, tempflag))
+				G.deactivate(M,0, tempflag)
+	M.alpha = 255
+	//M.mutations = list()
+	//M.active_genes = list()
+
+	M.disabilities = 0
+	M.sdisabilities = 0
+
+	//Makes it more obvious that it worked.
+	M.jitteriness = 0
+
+	// Might need to update appearance for hulk etc.
+	if(needs_update && ishuman(M))
+		var/mob/living/carbon/human/H = M
+		H.update_mutations()
+	..()
+	return
+
+/datum/chemical_reaction/mutadone
+	name = "Mutadone"
+	id = "mutadone"
+	result = "mutadone"
+	required_reagents = list("mutagen" = 1, "acetone" = 1, "bromine" = 1)
+	result_amount = 3
+
+
+/datum/reagent/mutadone
+	name = "Mutadone"
+	id = "mutadone"
+	description = "Heals your genetic defects."
+	color = "#C8A5DC"
+
+datum/reagent/antihol
+	name = "Antihol"
+	id = "antihol"
+	description = "Helps remove Alcohol from someone's body, as well as eliminating its side effects."
+	color = "#C8A5DC"
+
+datum/reagent/antihol/on_mob_life(var/mob/living/M as mob)
+	M.dizziness = 0
+	M.drowsyness = 0
+	M.slurring = 0
+	M.confused = 0
+	M.reagents.remove_reagent("ethanol", 8)
+	M.adjustToxLoss(-0.2*REM)
+	..()
+
+/datum/chemical_reaction/antihol
+	name = "antihol"
+	id = "antihol"
+	result = "antihol"
+	required_reagents = list("ethanol" = 1, "charcoal" = 1)
+	result_amount = 2
+
+/datum/chemical_reaction/cryoxadone
+	name = "Cryoxadone"
+	id = "cryoxadone"
+	result = "cryoxadone"
+	required_reagents = list("plasma" = 1, "acetone" = 1, "mutagen" = 1)
+	result_amount = 3
+
+/datum/reagent/stimulants
+	name = "Stimulants"
+	id = "stimulants"
+	description = "Increases run speed and eliminates stuns, can heal minor damage. If overdosed it will deal toxin damage and stun."
+	color = "#C8A5DC"
+	metabolization_rate = 0.4
+	overdose_threshold = 60
+	ignore_slowdown = 1
+
+datum/reagent/stimulants/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	if(M.health < 50 && M.health > 0)
+		if(prob(50))
+			M.adjustOxyLoss(-5*REM)
+			M.adjustToxLoss(-5*REM)
+			M.adjustBruteLoss(-5*REM)
+			M.adjustFireLoss(-5*REM)
+	M.adjustFireLoss(-3*REM)
+	M.AdjustParalysis(-1)
+	M.AdjustStunned(-1)
+	M.AdjustWeakened(-1)
+	..()
+
+datum/reagent/stimulants/overdose_process(var/mob/living/M as mob)
+	if(prob(33))
+		M.adjustToxLoss(3*REM)
+		M.losebreath++
+	..()
+	return
+
+datum/reagent/insulin
+	name = "Insulin"
+	id = "insulin"
+	description = "Increases sugar depletion rates."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+datum/reagent/insulin/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	if(M.sleeping)
+		M.sleeping--
```

Comment:
sugar overdose increases sleeping, this decreases it to help speed up the wake up when dosing a diabetic